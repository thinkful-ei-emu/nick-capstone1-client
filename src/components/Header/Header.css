



.title-header {
  color: white;
  font-size: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 160px;
  background-color: rgb(207, 55, 55);
  margin: 0 15%;
  padding: 0;
  border-bottom: 1px solid black;
  border-left: 1px solid black;
  border-right: 1px solid black;
}

.app-header {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  background-color: rgb(155, 163, 172);
  color: white;
  height: 100px;
  border-bottom: 1px solid black;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  margin: 0 15%;
  border-left: 1px solid black;
  border-right: 1px solid black;
}

.login-button {
  color: white;
  font-size: 18px;
  margin-left: 100px;
  min-height: 30px;
  width: 80px;
  background-color: rgb(207, 55, 55);
  border: 1px solid rgb(207, 55, 55);
  border-radius: 10px;
}

.login-button:hover{
  cursor: pointer;
}

a {
  text-decoration: none;
  color: black;
}

a :active{
  color: rgb(155, 163, 172);
}

.header-login {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  margin: 5% 5%;
}