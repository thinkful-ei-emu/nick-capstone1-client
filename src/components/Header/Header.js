import React from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/TokenService';
import HeaderContext from '../../contexts/HeaderContext';
import './Header.css'


class Header extends React.Component{
  

  static contextType = HeaderContext;

  
  
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    this.context.setAuthToken(TokenService.getAuthToken())
  }

  renderUserName = () => {
    if(this.context.authToken){
      return <h3 className='user-name'>{TokenService.getUserName()}</h3>
    }
    else {
      return <h3 className='user-name'>Guest</h3>
    }
  }

  renderLogoutLink() {

    return (
      <div className='header-login'>
       {this.renderUserName()}
        <Link to='/' onClick={this.handleLogoutClick}><button className='login-button'>Logout</button></Link>
      </div>
      )
  }

  renderLoginLink(){

  return  (
      <div className='header-login'>
       {this.renderUserName()}
        <Link to='/login'><button className='login-button'>Login</button></Link>
        {'  '}
        <Link to='signup'><button className='login-button'>Sign up</button></Link>
      </div>
    )
  }


  render(){
    return (
      <>
      <Link to={'/posts'}>
        <h1 className='title-header'>BandBridge</h1>
        </Link>
      <header className='app-header'>
        {this.context.authToken
        ? this.renderLogoutLink()
        : this.renderLoginLink()}
      </header>
      </>
    )
  }


}

export default Header;

