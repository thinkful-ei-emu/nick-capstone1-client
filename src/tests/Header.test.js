import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../components/Header/Header';
import {MemoryRouter} from 'react-router';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<MemoryRouter><Header /></MemoryRouter>, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders UI as expected', () => {
  const tree = renderer
      .create(<MemoryRouter><Header /></MemoryRouter>)
      .toJSON();
      expect(tree).toMatchSnapshot();
})