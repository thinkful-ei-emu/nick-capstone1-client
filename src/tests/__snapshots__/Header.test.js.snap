// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
Array [
  <a
    href="/posts"
    onClick={[Function]}
  >
    <h1
      className="title-header"
    >
      BandBridge
    </h1>
  </a>,
  <header
    className="app-header"
  >
    <div
      className="header-login"
    >
      <h3
        className="user-name"
      >
        Guest
      </h3>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="login-button"
        >
          Login
        </button>
      </a>
        
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="login-button"
        >
          Sign up
        </button>
      </a>
    </div>
  </header>,
]
`;
